src/text_shaper/directwrite_shaper.cpp

```diff
@@ -140,70 +140,84 @@ optional<font_key> directwrite_shaper::load_font(font_description const& _descri
     std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> wStringConverter;
     std::wstring familyName = wStringConverter.from_bytes(_description.familyName);
 
-    for (UINT32 i = 0, e = fontCollection->GetFontFamilyCount(); i < e; ++i)
-    {
-        IDWriteFontFamily* family{};
-        fontCollection->GetFontFamily(i, &family);
+    UINT32 familyIndex;
+    BOOL familyExists = FALSE;
+    fontCollection->FindFamilyName(familyName.data(), &familyIndex, &familyExists);
+
+    if (!familyExists) {
+        // Fallback to Consolas
+        const wchar_t* consolas = L"Consolas";
+        fontCollection->FindFamilyName(consolas, &familyIndex, &familyExists);
+    }
 
-        IDWriteLocalizedStrings* names;
-        family->GetFamilyNames(&names);
+    Microsoft::WRL::ComPtr<IDWriteFontFamily> fontFamily;
+    fontCollection->GetFontFamily(familyIndex, &fontFamily);
 
-        BOOL exists = FALSE;
-        unsigned index{};
-        names->FindLocaleName(d->userLocale.c_str(), &index, &exists);
+    Microsoft::WRL::ComPtr<IDWriteLocalizedStrings> familyNames;
```

Comment:
```suggestion
    ComPtr<IDWriteLocalizedStrings> familyNames;
```
should work, too. because it got imported on the top of the file.

Reply:
p.s.: can be changed at any time later. so I can merge early :)